app/Http/Requests/BaseApiRequest.php

```diff
@@ -200,7 +204,7 @@ protected function authorizeAlbumsWrite(BaseCollection $albums): bool
 	 */
 	protected function authorizeAlbumsWriteByIDs(array $albumIDs): bool
 	{
-		return $this->albumAuthorisationProvider->areEditableByIDs($albumIDs);
+		return Gate::check(UserPolicy::UPLOAD, User::class) && $this->albumPolicy->editById(Auth::user(), $albumIDs);
```

Comment:
Why have you added `Gate::check(UserPolicy::UPLOAD, User::class)`? This is confusing. If a user needs the 'upload'  capability to be authorized to edit a album, then this should be part of `editById`, but should not be necessary to do additional check besides that.

```suggestion
		return this->albumPolicy->editById(Auth::user(), $albumIDs);
```

BTW: `AlbumPolicy::editById` does indeed check for the upload capability.